backend/repositories/match-repository.js

```diff
@@ -103,10 +103,11 @@ exports.recordMatch = async (matchDescription) => {
     const newUsers = await storage.getAllUsers()
 
     try {
-        const bot = await botFactory.makeBot(process.env.FOOSBOT_TOKEN, process.env.FOOS_CHANNEL_NAME)
-        await bot.reportMatchOnSlack(match, oldUsers, newUsers)
+        if (bot) {
```

Comment:
Please switch `if` and `try`.

Reply:
the reason being to make the try scope as small as possible? fixed

Reply:
That too, but mainly so that you don't go into the "try mode" if it's not necessary...

It's like:
You: "Hey, program, try this and if you encounter this exception, do this."
Program: "Yeah, sure, what am I supposed to try?"
You: "Ummm... Well, actually the bot doesn't exist, so nothing after all."
Program: "WTF? Couldn't you have checked it first? I am all pumped up here and then nothing... Geez..."
:D